gitops/snapshot.go

```diff
@@ -125,6 +125,9 @@ var (
 // MarkSnapshotAsPassed updates the AppStudio Test succeeded condition for the Snapshot to passed.
 // If the patch command fails, an error will be returned.
 func MarkSnapshotAsPassed(adapterClient client.Client, ctx context.Context, snapshot *applicationapiv1alpha1.Snapshot, message string) (*applicationapiv1alpha1.Snapshot, error) {
+	if HaveAppStudioTestsSucceeded(snapshot) {
```

Comment:
I still think we could make this check in the adapter code before calling  `MarkSnapshotAsPassed()`.

Reply:
The reason for my suggestion is that the `MarkSnapshotAsPassed()` function implies that it will update the Snapshot with this status, and with this change it might just return the unmodified snapshot.

The adapter then reports the update event to the audit log (e.g. [here](https://github.com/redhat-appstudio/integration-service/blob/main/controllers/integrationpipeline/integrationpipeline_adapter.go#L143-L144)), even though the Snapshot might not have been updated at that time.

Reply:
If so, I am wondering if I need to add the check for all the places that MarkSnapshotAsPassed() are called. Such as  [here](https://github.com/redhat-appstudio/integration-service/blob/main/controllers/integrationpipeline/integrationpipeline_adapter.go#L138) in controllers/integrationpipeline/integrationpipeline_adapter.go?

Reply:
Yes, a new common function might be the way to go. If it can be made to generically handle different statuses, even better.

Reply:
Then in the code logic, before each of the MarkSnapshotAsPassed/ MarkSnapshotAsFailed/ SetSnapshotIntegrationStatusAsInvalid, we'll add the check that the snapshot has never been marked as "Succeeded/Failed/Invalid"  before.  Right?

Reply:
Thanks. @dirgim, shall we keep the checking inside the function ?

Reply:
Can we change the message to something like "Snapshot integration status condition marked/ kept as failed, some tests within Integration PipelineRuns failed"?

Reply:
I added the function of IsSnapshotStatusConditionSet() for checking the status of the snapshot and add the checking before the MarkSnapshotAsPassed/ MarkSnapshotAsFailed/ SetSnapshotIntegrationStatusAsInvalid.  Please help to review if it's as expected. Thanks

Reply:
This looks like a reasonable solution to me.